### ISSUE TITLE
Separation of concerns in KeywordsServices::Create

### ISSUE THREAD

**Issue by olivierobert** [VyTranSc/n_vy_asm#7]
When it comes of following the principle of single responsibility, we can hardly say that `KeywordsServices::Create` follows that to the T 😁 This class loops over all keywords, manages the Selenium session and Google search, and finally creates the keyword reports 💦 Each action could be in separate classes. At the minimum, the content searching and scraping should be in a separate class. What do you think?

On the testing side, `KeywordsServices::Create` contains the most business logic but [its resting coverage](https://github.com/VyTranSc/n_vy_asm/blob/master/spec/services/keywords_services/create_spec.rb) is rather minimal.

**Comment by VyTranSc** [comment]
Yes I agree that the `KeywordServices::Create` is doing multiple task in the same file which is not a very good way to organize the code and make it quite hard for me to write test for this service.

There was a commit where I start the Selenium and execute Google search action in a separate file to make the services easier to read and smaller to test (for example: https://github.com/VyTranSc/n_vy_asm/pull/4/commits/1fb96b5e8ce89bdeb051058008e57ffc8d4c52c5) but ended up merging those into one file to avoid opening new Chrome and quit it for each keyword, which make Chrome quite easy to crash or using too much resources and can make the cost higher if this is hosted in cloud services. Although I have to admit that the above commit still not fully organized yet but it's easier to test compare to the submitted one.
